cmake/std/trilinosprhelpers/setenvironment/SetEnvironment.py

```diff
@@ -0,0 +1,447 @@
+        """
+        status = 0
+        print("Module Operations")
+        print("+---------------+")
+        for m in self.actions["module-op"]:
+            op     = m[0]
+            params = m[1:]
+            cmd_status = 0
+            print("[module] module {} {}".format(op, " ".join(params)), end="")
+            if   0 == len(params) and op in ["purge"]:
+                cmd_status = max(ModuleHelper.module(op), status)
+            elif 1 == len(params):
+                cmd_status = max(ModuleHelper.module(op, params[0]), status)
+            elif 2 == len(params):
+                cmd_status = max(ModuleHelper.module(op, params[0], params[1]), status)
+            else:
+                print(" ERROR")
+                msg =  "Invalid number of parameters for `module` command.\n"
+                msg += "- Received {} parameters to a `{}` command.".format(len(params), op)
+                raise IndexError(msg)
+
+            status = max(cmd_status, status)
+
+            if 0 == cmd_status:
+                print(" OK")
+            else:
+                print(" FAILED")
+
+        print("")
+        print("Environment Vars")
+        print("+--------------+")
+
+        for envvar in self.actions["setenv"]:
+            k = envvar['key']
+            v = envvar['value']
+            print("[envvar] export {}={}".format(k,v))
+            v = self._expand_envvars_in_string(v)
+            os.environ[k] = v
+            print("[envvar] export {}={} (actual)".format(k,v))
+        for k in self.actions["unsetenv"]:
+            del os.environ[k]
+            print("[envvar] unset {}".format(k))
+
+        if status != 0 and throw_on_error:
+            sys.stdout.flush()
+            raise Exception("FATAL ERROR in SetEnvironment.apply()")
+
+        return status
+
+
+    def pretty_print_envvars(self, envvar_filter=None, filtered_keys_only=False):
+        """
+        Print out a filtered list of environment variables. This routine provides
+        a simplified view of the environment variables on a system since sometimes
+        just printing out the contents of all envvars on a system can result in very
+        cluttered console logs that are difficult to read through.
+
+        `envvar_filter` contains a list of strings that is matched against envvar
+        key names, if the envvar key contains one of the strings in envvar_filter
+        then we print both the envvar and its value. Otherwise, we print just the
+        envvar name.
+
+        If we set filtered_keys_only to True then we ONLY print out envvars that
+        matched envvar_filter.
+
+        Arguments:
+            envvar_filter (list): a list of keys to print out the value.
+                all envvar values are printed if omitted.
+                Default: None
+            filtered_keys_only (bool): If true, we only display filtered keys.
+                If false, we display the keys of all keys.
```

Comment:
Swap the first `keys` in this line for `values`?

Reply:
nope, it is correct but not well written.

If `filtered_keys_only` == false then we will display the keys of all envvars.  if filtered_keys_only is true then we completely ignore printing any envvars that don't have a match.